src/platform/iotjs_module_pwm-linux-general.inl.h

```diff
@@ -46,51 +46,84 @@ void iotjs_pwm_initialize() {
   DDLOG("PWM initialize");
 }
 
-
-// Set PWM period.
-bool SetPwmPeriod(iotjs_pwm_reqdata_t* req_data) {
-  char path[PWM_PATH_BUFFER_SIZE] = { 0 };
-  if (snprintf(path, PWM_PATH_BUFFER_SIZE - 1, PWM_PIN_FORMAT,
-               PWM_DEFAULT_CHIP_NUMBER, req_data->pin) < 0) {
-    return false;
+// Generate device path for specified PWM device.
+// The path may include node suffix if passed ('enable', 'period', 'duty_cycle')
+// Pointer to a allocated string is returned, or null in case of error.
+static char* generateDeviceSubpath(iotjs_pwm_reqdata_t* req_data,
+                                   const char* fileName) {
+  char* devicePath = NULL;
+  int size;
+  if ((size = asprintf(&devicePath, PWM_PIN_FORMAT, PWM_DEFAULT_CHIP_NUMBER,
```

Comment:
I used asprintf in order to avoid strncat/strncpy.
Those functions were not designed for what we need and the resulting code is ugly.
(https://www.sudo.ws/todd/papers/strlcpy.html)

The standard library we use (libc), has asprintf built-in for us to use.
Is adding compile-time guard enough, or you want the asprintf gone?